dvc/cache/base.py

```diff
@@ -141,31 +97,39 @@ def _do_link(self, from_info, to_info, link_method):
             "Created '%s': %s -> %s", self.cache_types[0], from_info, to_info,
         )
 
-    def add(self, path_info, tree, hash_info, **kwargs):
-        if not self.changed_cache_file(hash_info):
+    def get(self, hash_info):
+        """ get raw object """
+        return HashFile(
+            # Prefer string path over PathInfo when possible due to performance
```

Comment:
Very soon this will no longer be needed, when we detach trees from their self.path_info's.